src/judoscale/core/reporter.py

```diff
@@ -38,10 +62,12 @@ def _build_report(self, metrics):
         return {
             "dyno": config.dyno,
             # TODO: Does each thread get its own PID?
-            "pid": os.getpid(),
+            "pid": self.get_pid,
             "config": config.for_report(),
             "metrics": list(map(metric_to_list, metrics)),
         }
 
 
+import signal
 reporter = Reporter()
+signal.signal(signal.SIGILL, reporter.signal_handler)
```

Comment:
this signal will be used if we want to terminate the thread with a KILL process.

Reply:
Good idea! Heroku [terminates dynos with SIGTERM](https://devcenter.heroku.com/articles/dynos#graceful-shutdown-with-sigterm), so I think we should handle that instead of (or in addition to) SIGKILL. wdyt?

Also, I assume `SIGILL` is a typo?

Reply:
uh yes, anyhow I removed it and replaced it with SIGTERM since you say this is what is used  by heroku. Locally threads are terminated with ctr+C (SIGINT) when we exit the local runserver which by default kills all daemon threads.